vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridListDataView.java

```diff
@@ -32,20 +34,24 @@
  */
 public class GridListDataView<T> extends AbstractListDataView<T>
         implements GridDataView<T> {
+    private SerializableSupplier<DataCommunicator<T>> dataCommunicatorSupplier;
+    private SerializableSupplier<Grid<T>> gridSupplier;
 
-    /**
-     * Construct a new GridListDataView.
-     *
-     * @param grid
-     *         DataView Grid instance
-     */
-    public GridListDataView(Grid grid) {
-        super(new GridDataController<>(grid));
+    public GridListDataView(
+            SerializableSupplier<DataCommunicator<T>> dataCommunicatorSupplier,
+            SerializableSupplier<Grid<T>> gridSupplier) {
+        super(() -> dataCommunicatorSupplier.get().getDataProvider(),
+                gridSupplier);
+        this.dataCommunicatorSupplier = dataCommunicatorSupplier;
+        this.gridSupplier = gridSupplier;
```

Comment:
Like commented on the flow PR, these two (component & communicator) will stay the same while data provider can change so it looks like the supplier is overkill for these

Reply:
done